spec/facets/DatasetProviderDatasetFacet.json

```diff
@@ -0,0 +1,37 @@
+{
+  "$schema" : "https://json-schema.org/draft/2020-12/schema",
+  "$id" : "https://openlineage.io/spec/facets/1-0-0/DatasetProviderDatasetFacet.json",
+  "$defs" : {
+    "DatasetProviderDatasetFacet": {
+      "allOf": [
+        {
+          "$ref": "https://openlineage.io/spec/1-0-2/OpenLineage.json#/$defs/DatasetFacet"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "datasetProvider": {
```

Comment:
Since we are already in the "provider" facet, this would be "name".
We should document a list of known providers to ensure consistency. Would that apply to other things than Table Formats? Possibly a name more specific than "provider" would be appropriate.

Reply:
Given the discussion we have, term `provider` seems to be kind of misleading. 
Official iceberg and delta definitions are:
 * `Iceberg is a high-performance format for huge analytic tables`
 * `Delta Lake is a storage layer that brings scalable, ACID transactions to Apache Spark and other big-data engines.`

Following that, we could name it `storage` of the type `DatasetStorageDatasetFacet` with the fields:
 * `storageLayer` which can be `iceberg`, `delta`
 * `fileFormat` which can be `parquet`, `orc`, etc. 
 
Alternatively we could go with `StorageLayerDatasetFacet` which is also OK but has a limited extensibility in the future.

Reply:
I modified facet into `DatasetStorageDatasetFacet` as it describes better its content.

Reply:
Thanks for the comments. I've applied the changes accordingly.